### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by jorgebay**
- Add eslint config file
- lint lib source files
- Run eslint as part of the Travis build

Exclude `integer.js` from linting as we are not the authors.

- [x]  Used #197 as base branch to avoid conflicts on `lib/request-handler.js`, I will rebase once merged.

### CODE REVIEW

**Comment by tolbertam** [comment]
Great idea, I think we should either merge this first and then rebase/update all the other branches, or merge all the other branches first and then rebase and apply this.  wdyt?

**Comment by jorgebay** [comment]
We can rebase it after #197 is merged.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
   - "0.10"
-  - "iojs"
 sudo: false
 cache:
   directories:
     - node_modules
+script:
+  - if [ "$TRAVIS_NODE_VERSION" != "0.10" ] && [ "$TRAVIS_NODE_VERSION" != "0.12" ]; then
+      npm install -g eslint;
+      eslint lib --ignore-pattern "/lib/types/integer.js";
```
Comment (tolbertam) [inline .travis.yml]: should we lint tests too?  Or maybe should we save that for another PR?  (I can do the work for that if you think we should)

Comment (jorgebay) [inline .travis.yml]: We should enable linting for tests too, this pr is only a first step, we can do it in another pr.
